internal/pkg/client/git/client_test.go

```diff
@@ -19,18 +19,110 @@
 package git_test
 
 import (
+	"context"
+	"os"
+	"path/filepath"
 	"testing"
 
-	"github.com/stretchr/testify/assert"
+	gogit "github.com/go-git/go-git/v5"
+	"github.com/protobom/protobom/pkg/reader"
+	"github.com/protobom/protobom/pkg/sbom"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/suite"
 
 	"github.com/bomctl/bomctl/internal/pkg/client/git"
+	"github.com/bomctl/bomctl/internal/pkg/db"
+	"github.com/bomctl/bomctl/internal/pkg/options"
 	"github.com/bomctl/bomctl/internal/pkg/url"
 )
 
-func TestParse(t *testing.T) {
+var TestDataDir = filepath.Join("..", "..", "db", "testdata")
+
+type gitSuite struct {
+	suite.Suite
+	tempDir string
+	repo    *gogit.Repository
+	opts    *options.Options
+	backend *db.Backend
+	gc      *git.Client
+	doc     *sbom.Document
+	docs    []*sbom.Document
+}
+
+func (gs *gitSuite) SetupSuite() {
+	dir, err := os.MkdirTemp("", "testrepo")
+	if err != nil {
+		gs.T().Fatalf("Failed to create temporary directory: %v", err)
+	}
+
+	gs.tempDir = dir
```

Comment:
Do we have a plan/place where we do a teardown, especially if we encounter an error to remove the temp directory?

Same as above

https://pkg.go.dev/os#MkdirTemp

> It is the caller's responsibility to remove the directory when it is no longer needed.

`defer os.RemoveAll(dir) // clean up`

Reply:
Called during the TearDown Suite Function